src/components/atom/IconButton/IconButton.tsx

```diff
@@ -0,0 +1,13 @@
+import { IconButtonProps } from './types'
+import { IconButtonStyled } from './elements'
+
+
+const IconButton = (props: IconButtonProps) => {
```

Comment:
A better way of destructuring would be doing it here like this
`const IconButton = ({children, ...props}: IconButtonProps) => {`

This way, the `children` props will be excluded from the `IconButtonStyled` props since you only need `children` props in the child element.

Reply:
to clarify by using the following format `const IconButton = ({children, ...props}: IconButtonProps) => {` results in distructuring "children" from props and spreading the rest into the functional component? I would still need to do `<IconButtonStyled {...props}>` right?

Reply:
@Maurice63 Yup! You still need that part since you're using the `onclick` and the other props for styling right? Don't forget to remove this part `const {children} = props;`.

You can also destructure everything to make it less confusing and manually add the props one by one like this
```
const IconButton = ({children, onClick, disabled}: IconButtonProps) => {
...
return (
   <IconButtonStyled
      onClick={onClick}
      disabled={disabled}
   >
      {children}
   <IconButtonStyled>
...
```
This is more preferred than the spread operator since it's clearer. You may need to adjust your `elements.ts` file though since some of your props are not optional.